### DISCUSSION TITLE
Could not embed because more than one relationship was found for 'spots' and 'testimonial_pages'

### DISCUSSION THREAD

**Discussion by YossHar** [supabase/supabase#29308] [Questions]
I am trying to fetch 'spots' table row that has 'testimonial_pages' fkey.
Only spots have fkey to 'testimonial_pages'.
testimonial_pages had fkey to spots but I deleted him.

```
const query = await supabaseClient
        .from('spots')
        .select(`id, name, created_at, testimonial_page_id, testimonial_page: testimonial_pages(*)`)
        .eq('id', id)
        .single();
```

And I get error:
```
{
    "code": "PGRST201",
    "details": [
        {
            "cardinality": "one-to-many",
            "embedding": "spots with testimonial_pages",
            "relationship": "testimonial_pages_spot_id_fkey using spots(id) and testimonial_pages(spot_id)"
        },
        {
            "cardinality": "many-to-one",
            "embedding": "spots with testimonial_pages",
            "relationship": "spots_testimonial_page_id_fkey using spots(testimonial_page_id) and testimonial_pages(id)"
        }
    ],
    "hint": "Try changing 'testimonial_pages' to one of the following: 'testimonial_pages!testimonial_pages_spot_id_fkey', 'testimonial_pages!spots_testimonial_page_id_fkey'. Find the desired relationship in the 'details' key.",
    "message": "Could not embed because more than one relationship was found for 'spots' and 'testimonial_pages'"
}
```

Running:
`NOTIFY pgrst, 'reload schema'`
Did not work.

**Comment by GaryAustin1** [comment]
Maybe show your table definitions (in particular the fk constraints.  You can get them with the definition tab in the table UI.

**Reply by YossHar** [reply]
![Screenshot 2024-09-16 at 21 02 24](https://github.com/user-attachments/assets/3ce9f5a4-cdc2-4ad8-9b24-0ab9c7428e96)

**Reply by GaryAustin1** [reply]
That is not the table definition (selected from bottom left).  But if nothing is messed up with the UI interpreting it that side looks good.

What about the testimonial_pages table as that one also seems to have an fk back to spots table.  The error is saying both exists and you have to pick one.

Edit: and you would normally not have fk from each table to the other.

**Reply by YossHar** [reply]
Got it.
Testimonials table shows I have fk here. but in the data view - fk disappears.
![Screenshot 2024-09-16 at 21 17 56](https://github.com/user-attachments/assets/8484c13f-a21f-4182-816e-0bd8930f5bdc)
![Screenshot 2024-09-16 at 21 19 57](https://github.com/user-attachments/assets/beaffab9-d1a9-4bf9-abf3-aaf768ab1773)

**Reply by GaryAustin1** [reply]
I think you are saying the UI shows another fk, but table definition does not.
Try restarting your instance.
